src/main/java/org/jabref/gui/groups/GroupDialogViewModel.java

```diff
@@ -335,10 +335,12 @@ public AbstractGroup resultConverter(ButtonType button) {
                             FieldFactory.parseField(autoGroupPersonsFieldProperty.getValue().trim()));
                 }
             } else if (typeTexProperty.getValue()) {
+                // issue 7719: add workingDir to filepath if it is relative
+                Path inputPath = preferencesService.getWorkingDir().resolve(Path.of(texGroupFilePathProperty.getValue().trim()));
                 resultingGroup = TexGroup.create(
                         groupName,
                         groupHierarchySelectedProperty.getValue(),
-                        Path.of(texGroupFilePathProperty.getValue().trim()),
```

Comment:
Thank you for looking at the issue!

Perhaps the changes should be added with 
https://github.com/JabRef/jabref/blob/807f85e4ab418b4398408e37f023ee09209b038a/src/main/java/org/jabref/model/groups/TexGroup.java#L134-L143

instead. Wouldn't they break paths relative to the "latex directory" otherwise? (refs https://github.com/JabRef/jabref/pull/4792)

Reply:
Hi, thanks for your suggestion. Do you mean that it is better to add the  `preferencesService.getWorkingDir()` to file path  in`expandPath(Path path)` method instead of in jabref/src/main/java/org/jabref/gui/groups/GroupDialogViewModel.java. Could you explain it a little further?

Reply:
> In my modification, `this.filePath` is always absolute. I thinks it is ok with `FileHelper.find(path.toString(), fileDirectories).orElse(path); ` .

Reply:
Quite frankly I am not really sure what goes on here at all. I am not sure what I think of `getWorkingDir()`, but the "working dir" is changed if you open a different library which is strange behavior.

I think you are right about it not breaking the "latex directory". There is a bug in the `texGroupFilePathValidator` that makes it seem to not work as I'd expect anyway (that is unrelated to your changes).

> Do you mean that it is better to add the `preferencesService.getWorkingDir()` to file path in`expandPath(Path path)` method instead of in jabref/src/main/java/org/jabref/gui/groups/GroupDialogViewModel.java. Could you explain it a little further?

I think so. I'll try to think a bit more about it, but I would prefer if someone in the future will be able to use the `TexGroup.expandPath` method (or a similar method) to fix the bug in `texGroupFilePathValidator`. That the path to a valid absolute or relative `.aux`-file is created in the same method (DRY).

Reply:
What are your thoughts about using the `currentDatabase.getDatabasePath()`?

Reply:
I print "working dir"  out and it always turns out to be the home dir for current user(Mac OS). 

I have tried some input and find that `currentDatabase.getDatabasePath()` is empty. I am not quite sure if `metaData.getLatexFileDirectory(user)` and  `currentDatabase.getDatabasePath()` are the same.  I think explicitly setting `currentDatabase.getDatabasePath()` to working dir doesn't sound good as well.

There is also another way to fix it. Remove the automatically added "working dir" in `texGroupFilePathValidator` and always let users input a absolute path or use the browser button. This can make it simple, since probably `getWorkingDir()`'s behaviour is wired.

Reply:
> I am not quite sure if `metaData.getLatexFileDirectory(user)` and `currentDatabase.getDatabasePath()` are the same.

- `getLatexFileDirectory` can be manually set by the user and is intended to allow relative paths to a user-specified directory (the menu `Library` -> `Library properties`) for `.aux` files
- `getDatabasePath` _should_ give you the path to the .bib file. It should be used by the "Reveal in file explorer" function. (From the JavaDoc) `Get the path where this database was last saved to or loaded from, if any.`


> I think explicitly setting `currentDatabase.getDatabasePath()` to working dir doesn't sound good as well.

Agreed.

> There is also another way to fix it. Remove the automatically added "working dir" in `texGroupFilePathValidator` and always let users input a absolute path or use the browser button. This can make it simple, since probably `getWorkingDir()`'s behaviour is wired.

I think you are right in that the "working dir" should be removed from the `texGroupFilePathValidator`. Perhaps its behavior can be made more aligned with
https://github.com/JabRef/jabref/blob/84c8849e045d1568a4f088dc09bc51594a367d61/src/main/java/org/jabref/gui/groups/GroupDialogViewModel.java#L422-L431

It seems like there is a wish to be able to use relative paths, which were the intent of the latex file directory. What are your thoughts about attempting to get it working again and updating the documentation?
I don't know if there is a "default" directory it would make sense to look for `.aux` files in, but perhaps it isn't too much to ask of a user to specify that directory.

Reply:
I have set the base for relative filePath as `getLatexFileDirectory()` and I have try some test cases manually and it can work. Point it out if there is any problem.